meteor/server/api/blueprints/context/adlibActions.ts

```diff
@@ -175,6 +212,17 @@ export class ActionExecutionContext extends ShowStyleUserContext implements IAct
 		}
 	}
 
+	getPartForPreviousPiece(piece: UnprotectedStringProperties<Pick<Piece, '_id'>>): IBlueprintPart | undefined {
+		if (!piece?._id) {
+			throw new Error('Cannot find Part from invalid Piece')
+		}
+
+		const pieceDB = Pieces.findOne({ _id: protectString(piece._id) })
```

Comment:
this should have a check on the startRundownId to ensure it belongs to a rundown in the playlist